cudapoa/src/application_parameters.cpp

```diff
@@ -0,0 +1,225 @@
+            max_windows = std::stoi(optarg);
+            break;
+        case 'R':
+            gpu_mem_allocation = std::stod(optarg);
+            break;
+        case 'm':
+            match_score = std::stoi(optarg);
+            break;
+        case 'g':
+            gap_score = std::stoi(optarg);
+            break;
+        case 'n':
+            mismatch_score = std::stoi(optarg);
+            break;
+        case 'v':
+            print_version();
+        case 'h':
+            help(0);
+        default:
+            exit(1);
+        }
+    }
+
+    if (gpu_mem_allocation <= 0 || gpu_mem_allocation > 1.0)
+    {
+        std::cerr << "gpu-mem-alloc should be greater than 0 and less than 1.0" << std::endl;
+        exit(1);
+    }
+
+    if (banded && band_width < 1)
+    {
+        std::cerr << "band-width must be positive" << std::endl;
+        exit(1);
+    }
+
+    if (result < 0 || result > 1)
+    {
+        std::cerr << "result can only be 0 (consensus) or 1 (msa)" << std::endl;
+        exit(1);
+    }
+
+    if (match_score < 0)
+    {
+        std::cerr << "match score must be positive" << std::endl;
+        exit(1);
+    }
+
+    if (max_windows == 0)
+    {
+        std::cerr << "max-windows cannot be 0" << std::endl;
+        exit(1);
+    }
+
+    if (mismatch_score > 0)
+    {
+        std::cerr << "mismatch score must be non-positive" << std::endl;
+        exit(1);
+    }
+
+    if (gap_score > 0)
+    {
+        std::cerr << "gap score must be non-positive" << std::endl;
+        exit(1);
+    }
+
+    verify_input_files(input_paths);
+}
+
+void ApplicationParameters::verify_input_files(std::vector<std::string>& input_paths)
+{
+    // Checks if the files are either all fasta or if one file is provided, it needs to be fasta or cudapoa
+    all_fasta = true;
+    for (auto& file_path : input_paths)
+    {
+        std::ifstream infile(file_path.c_str());
+        if (infile.good())
+        {
+            std::string firstLine;
+            std::getline(infile, firstLine);
+            if (firstLine.at(0) != '>')
+                all_fasta = false;
+            //break; TODO- Break here or not? It also provides input filepath verification @atadkase
+        }
+        else
+        {
+            std::cerr << "Invalid input file: " << file_path << std::endl;
+            exit(1);
+        }
+    }
+    if (input_paths.size() == 0 || (!all_fasta && input_paths.size() > 1))
+    {
+        std::cerr << "Invalid input. cudapoa needs input in either one cudapoa format file or in one/multiple fasta files." << std::endl;
+        help(1);
+    }
+}
+
+void ApplicationParameters::print_version(const bool exit_on_completion)
+{
+    std::cerr << claraparabricks_genomeworks_version() << std::endl;
+
+    if (exit_on_completion)
+    {
+        exit(1);
+    }
+}
+
+void ApplicationParameters::help(int32_t exit_code)
+{
+    std::cerr <<
+        R"(Usage: cudapoa [options ...]
+     options:
+        -i, --input <file>
```

Comment:
Not sure if I should leave this as input with `-i` or whether I should interpret any following strings as names for sequence files

Reply:
If multiples files, okay to have this syntax `cudapoa -i file1.fa -i file2.fa -i file3.fa ...`